### PR TITLE
[Service Connector] `az containerapp connection create`: Add required parameter for interactive mode

### PR DESCRIPTION
**Pull request by ChenTanyi**
**Related command**

`az containerapp connection create`

**Description**

* Add `--container` parameter in `az containerapp connection create` interactive mode.
* Change default client_type to `none`

**Testing Guide**

**History Notes**

[Service Connector] BREAKING CHANGE: `az containerapp connection create`: Default client_type changed to `none`
[Service Connector] `az containerapp connection create`: Add `--container` parameter in interactive mode

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/serviceconnector/_resource_config.py [hunk 1]
```diff
@@ -183,6 +183,18 @@ class CLIENT_TYPE(Enum):
 }
 
 
+# The dict defines the required parameters used in the source resources for creation.
+SOURCE_RESOURCES_CREATE_PARAMS = {
```
Comment (houk-ms) [inline src/azure-cli/azure/cli/command_modules/serviceconnector/_resource_config.py:187]: can we put it in `SOURCE_RESOURCES_PARAMS`? what's the impact? will it bring more complexity in interactive exp implementation?

File: src/azure-cli/azure/cli/command_modules/serviceconnector/_validators.py [hunk 2]
```diff
@@ -384,6 +385,19 @@ def get_missing_source_args(cmd):
 
     return missing_args
 
+def get_missing_source_create_args(cmd, namespace):
```
Comment (houk-ms) [inline src/azure-cli/azure/cli/command_modules/serviceconnector/_validators.py:388]: just a note, may be not the scope of this PR. 
the logic for interactive experience is getting more and more confusing now, need refactoring.

File: src/azure-cli/azure/cli/command_modules/serviceconnector/custom.py [hunk 3]
```diff
@@ -307,7 +308,7 @@ def connection_update(cmd, client,  # pylint: disable=too-many-locals
         },
         'client_type': client_type or linker.get('clientType'),
         # scope does not support update due to aks solution's limitation
```
Comment (houk-ms) [inline src/azure-cli/azure/cli/command_modules/serviceconnector/custom.py:310]: comment need update

**Review by houk-ms — APPROVED** [review]
overally lgtm

File: src/azure-cli/azure/cli/command_modules/serviceconnector/custom.py (hunk 3 above)
Comment (ChenTanyi) [reply src/azure-cli/azure/cli/command_modules/serviceconnector/custom.py]: changed

**Comment by yonzhan** [comment]
Service Connector

**Comment by ChenTanyi** [comment]
@kairu-ms please help to review and merge the PR. Thanks!

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**